modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/nhttp/NHttpTransportListenerTest.java

```diff
@@ -0,0 +1,175 @@
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*  http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing,
+*  software distributed under the License is distributed on an
+*  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*  KIND, either express or implied.  See the License for the
+*  specific language governing permissions and limitations
+*  under the License.
+*/
+package org.apache.synapse.transport.nhttp;
+
+import junit.framework.Assert;
+import org.apache.axiom.om.util.AXIOMUtil;
+import org.apache.axis2.addressing.EndpointReference;
+import org.apache.axis2.context.ConfigurationContext;
+import org.apache.axis2.context.MessageContext;
+import org.apache.axis2.context.OperationContext;
+import org.apache.axis2.context.ServiceContext;
+import org.apache.axis2.context.SessionContext;
+import org.apache.axis2.description.InOutAxisOperation;
+import org.apache.axis2.description.Parameter;
+import org.apache.axis2.description.TransportInDescription;
+import org.apache.axis2.description.TransportOutDescription;
+import org.apache.axis2.engine.AxisConfiguration;
+import org.apache.axis2.engine.AxisEngine;
+import org.apache.commons.httpclient.HttpClient;
+import org.apache.commons.httpclient.methods.PostMethod;
+import org.apache.commons.httpclient.methods.StringRequestEntity;
+import org.apache.synapse.transport.passthru.util.RelayUtils;
+import org.apache.synapse.transport.utils.ServiceUtils;
+import org.apache.synapse.transport.utils.TCPUtils;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import static org.mockito.ArgumentMatchers.any;
+
+/**
+ * Test class for nhttp Transport Listener.
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(AxisEngine.class)
+@PowerMockIgnore("javax.management.*")
+public class NHttpTransportListenerTest {
+    private static final String HOST = "127.0.0.1";
+    private static final int PORT = 8486;
+    private static HttpCoreNIOListener nHttpListener = new HttpCoreNIOListener();
+    private static TransportInDescription transportInDescription = new TransportInDescription("http");
+
+    @BeforeClass()
+    public static void startListener() throws Exception {
+        Assert.assertFalse("Port already occupied. Can not execute test", TCPUtils.isPortOpen(PORT, HOST));
+        Parameter portParam = new Parameter("port", PORT + "");
```

Comment:
Empty string has been concatenated? Is it intentional?

Reply:
Yes. HttpCoreNIOListener accept the port parameter as a string.